service/controller/v12/resource/migration/resource.go

```diff
@@ -61,30 +57,17 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	if err != nil {
 		return microerror.Mask(err)
 	}
-	customObject = *customObject.DeepCopy()
 
-	oldSpec := *customObject.Spec.DeepCopy()
-
-	if reflect.DeepEqual(providerv1alpha1.AWSConfigSpecAWSHostedZones{}, customObject.Spec.AWS.HostedZones) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "CR is missing hosted zone names")
-
-		apiDomain := customObject.Spec.Cluster.Kubernetes.API.Domain
-		zone, err := zoneFromAPIDomain(apiDomain)
-		if err != nil {
-			return microerror.Mask(err)
-		}
-
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("settings all zones to %q", zone))
-
-		customObject.Spec.AWS.HostedZones.API.Name = zone
-		customObject.Spec.AWS.HostedZones.Etcd.Name = zone
-		customObject.Spec.AWS.HostedZones.Ingress.Name = zone
-	}
-
-	if reflect.DeepEqual(customObject.Spec, oldSpec) {
+	// If credential secret is set we have nothing to migrate.
+	if customObject.Spec.AWS.CredentialSecret.Name != "" {
```

Comment:
Enables migration resource with logic from v11. This lets us upgrade directly to v12.

@TheoBrigitte Does this look OK?

Reply:
Looks all fine

Reply:
Cool thanks! 👍